data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3Source.java

```diff
@@ -99,6 +103,12 @@ public void start(Buffer<Record<Event>> buffer) {
         sqsService = new SqsService(acknowledgementSetManager, s3SourceConfig, s3Service, pluginMetrics);
 
         sqsService.start();
+        final S3ScanService s3ScanService;
+        if(s3ScanScanOptional.isPresent()) {
+            s3ScanService = new S3ScanService(s3SourceConfig,buffer,bucketOwnerProvider,new EventMetadataModifier(
```

Comment:
Please pass in the `S3ObjectHandler` here as well. I have a related comment below.